modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/ldap/LdapAuthHandler.java

```diff
@@ -0,0 +1,158 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package io.ballerina.messaging.broker.auth.ldap;
+
+import io.ballerina.messaging.broker.auth.AuthException;
+import io.ballerina.messaging.broker.auth.BrokerAuthConfiguration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.Hashtable;
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+/**
+ * Handles Ldap queries.
+ */
+public class LdapAuthHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LdapAuthHandler.class);
+    private BrokerAuthConfiguration.LdapConfiguration ldapConfiguration;
+
+    public LdapAuthHandler(BrokerAuthConfiguration.LdapConfiguration ldapConfiguration) throws AuthException {
+
+        this.ldapConfiguration = ldapConfiguration;
+
+        if (ldapConfiguration.getSsl().isEnabled()) {
+
+            String trustStoreLocation = Paths.get(ldapConfiguration.getSsl().getTrustStore().getLocation())
+                    .toAbsolutePath().toString();
+            try (FileInputStream fin = new FileInputStream(trustStoreLocation)) {
+
+                KeyStore trustStore = KeyStore.getInstance(ldapConfiguration.getSsl().getTrustStore().getType());
+                trustStore.load(fin, ldapConfiguration.getSsl().getTrustStore().getPassword().toCharArray());
+                TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
+                        ldapConfiguration.getSsl().getTrustStore().getCertType());
+                trustManagerFactory.init(trustStore);
+                SSLContext sslContext = SSLContext.getInstance(ldapConfiguration.getSsl().getProtocol());
+                sslContext.init(null, trustManagerFactory.getTrustManagers(), null);
+                LdapSslSocketFactory.setSslContext(sslContext);
+
+            } catch (KeyStoreException | IOException | NoSuchAlgorithmException | CertificateException
+                    | KeyManagementException e) {
+                throw new AuthException("Initializing ldap trust store failed", e);
+            }
+        }
+    }
+
+    public DirContext connectAnonymously() throws NamingException {
+
+        return connect(new Hashtable<String, String>());
```

Comment:
Explicit type argument String, String can be replaced with <> - SonarLint

Reply:
ok